src/DriverFactory.php

```diff
@@ -93,6 +93,7 @@ protected function getDefaultConfig(string $configKey) : array
             'paths' => [],
             'extension' => null,
             'drivers' => [],
+            'default_driver' => null,
```

Comment:
I'd say that relying on the end user providing a full config is quite risky: maybe the defaults in the config should be merged with user-provided config?

Reply:
Contrarily, `getDefaultConfig()` here guarantees that if any of the keys missing in the config provided by user, make them available on merged result generated at `AbstractFactory::retrieveConfig`.

I wonder if I really missed something here.

Reply:
Nay, that's fine, I was mostly confused by the defaults having `example/` in the path name